devel/start-localhost-registry

```diff
@@ -18,5 +23,7 @@ docker run \
     --publish 127.0.0.1:"$PORT":"$PORT" \
     --restart always \
     --name "$NAME" \
+    --volume "$DATA":/var/lib/registry \
```

Comment:
_(non-blocking)_

Upon `make clean` I noticed that the new `build/` directory had a size of 2.1G. I tried to find the accumulated volumes from previous registry usage, but none had large sizes except for the builder volume which is used elsewhere. Building from `master` also does not produce any new volumes in `docker volume ls`.

I think I'm missing something here. Without the custom `--volume`, do the transient volumes get pruned somehow?

Reply:
Hmm. They don't get pruned for me, hence the reason for the change, and I can see them in `docker volume ls`. On `master`, for example:

    $ docker volume ls 
    DRIVER    VOLUME NAME

    $ ./devel/start-localhost-registry 

    $ docker volume ls 
    DRIVER    VOLUME NAME
    local     85e2e9bfedc4a83eea0f00b978133d37b26fbd020294143b40f9c50d32291d87

    $ docker container inspect nextstrain-local-registry -f '{{json .Mounts}}' | jq
    [
      {
        "Type": "volume",
        "Name": "85e2e9bfedc4a83eea0f00b978133d37b26fbd020294143b40f9c50d32291d87",
        "Source": "/var/lib/docker/volumes/85e2e9bfedc4a83eea0f00b978133d37b26fbd020294143b40f9c50d32291d87/_data",
        "Destination": "/var/lib/registry",
        "Driver": "local",
        "Mode": "",
        "RW": true,
        "Propagation": ""
      }
    ]

    $ ./devel/stop-localhost-registry 

    $ docker volume ls 
    DRIVER    VOLUME NAME
    local     85e2e9bfedc4a83eea0f00b978133d37b26fbd020294143b40f9c50d32291d87

    $ docker container inspect nextstrain-local-registry -f '{{json .Mounts}}' | jq
    Error: No such container: nextstrain-local-registry

Perhaps something is auto-pruning them for you, e.g. Docker Desktop or a newer Docker version?

Reply:
What does `docker system df` report for you?

Reply:
Oh nevermind, I see the 2GB volume now. I think Docker Desktop's UI was displaying an incorrect size, or I misread earlier. 

I can reproduce the Docker Desktop UI to show these incorrect sizes after a change in "status":

![image](https://github.com/nextstrain/docker-base/assets/13424970/5d28028a-783c-425e-9514-1e5a8e11d996)

Clicking out then back into the "Volumes" view shows numbers that better reflect what's reported by `docker system df -v`:

```
Local Volumes space usage:

VOLUME NAME                                                        LINKS     SIZE
3647c4b5c1767b95a75be5bcc79ede3ff2cce3e9b52759361f6b3cd3cc0cb96d   0         0B
81f2fbdfd9fa34492dfb49e6761ad50f845c380275cd4a5267276cdc23a39e85   0         2.13GB
buildx_buildkit_nextstrain-builder0_state                          1         5.456GB
d99f38bd772f04ead379fb7eadc20de6589e9c66dd64d3eb3ad605f2c36dd63f   0         0B
f974ecaf61113fe8596ba6bcceaadb364a7435e071a060383fac9e1934030097   0         0B
823334b73184b6465a727efd7835a373a0426650e734a9f4a7c5a85a5ac11f59   0         2.13GB
c0c7d34820a0ebb85914fcbdfe7d6981ff3877ee394866bf6476df506b18a4b1   1         0B
3002016aaf02647af6150de06e83ceec9544bfb8a207a94f801dfe5506764f81   0         0B
fc3b5d46e5200e3ae2ccf1302e1b2c2780a13a17f8cf80880eda6b3519dcc11f   0         0B
```

Reply:
:upside_down_face: